clusterctl/clusterdeployer/clusterclient/clusterclient.go

```diff
@@ -46,6 +46,27 @@ const (
 	timeoutResourceDelete       = 15 * time.Minute
 )
 
+// Provides interaction with a cluster
+type Client interface {
```

Comment:
From https://github.com/golang/go/wiki/CodeReviewComments#interfaces:

"Go interfaces generally belong in the package that uses values of the interface type, not the package that implements those values."

This interface should stay in the clusterdeployer package, since that's what uses the interface. This package should just use the concrete type.

Reply:
This is interesting.  I've never seen any go project that does it this way, including Kubernetes.  It also go against decades of C++ for me.

Based on these guidelines, how would I handle the ClientFactory?  The interface is used in multiple packages.  If I define the interface where it's used, I'd have to create the interface in each of the packages.